File: List_1201/zzx/2.py
#顺序遍历较为简单 新建listnode的方法如*
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        newhead=ListNode(0)
        newpoint=newhead
        p1,p2=l1,l2
        while p1 or p2:
            a=p1.val if p1 else 0
            b=p2.val if p2 else 0
            result=(a+b+newpoint.val)%10
            flag=(a+b+newpoint.val)//10
            if p1:
                p1=p1.next
            if p2:
                p2=p2.next
            newpoint.val=result
            if p1 or p2 or flag>0:
                newpoint.next=ListNode(flag)#*
                newpoint=newpoint.next
        return newhead

File: List_1201/jiqing/2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode ans = new ListNode(0);
        ListNode tail = ans;
        int sum = 0;
        while(l1 != null || l2 != null || sum > 0) {
            sum += (l1 == null ? 0 : l1.val) + (l2 == null ? 0 : l2.val);
            tail.next = new ListNode(sum % 10);
            tail = tail.next;
            sum /= 10;
            if(l1 != null) l1 = l1.next;
            if(l2 != null) l2 = l2.next;
        }
        return ans.next;
    }
}